purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/components/common/ComponentOverride.kt

```diff
@@ -0,0 +1,68 @@
+package com.revenuecat.purchases.paywalls.components.common
+
+import com.revenuecat.purchases.InternalRevenueCatAPI
+import com.revenuecat.purchases.common.warnLog
+import com.revenuecat.purchases.paywalls.components.PartialComponent
+import dev.drewhamilton.poko.Poko
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.SerializationException
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.descriptors.buildClassSerialDescriptor
+import kotlinx.serialization.descriptors.element
+import kotlinx.serialization.encoding.CompositeDecoder
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+
+@InternalRevenueCatAPI
+@Poko
+@Serializable
+class ComponentOverride<T : PartialComponent>(
+    @get:JvmSynthetic val conditions: List<Condition>,
+    @get:JvmSynthetic val properties: T,
+) {
+
+    @Serializable(with = ConditionSerializer::class)
+    enum class Condition {
+        COMPACT,
+        MEDIUM,
+        EXPANDED,
+        INTRO_OFFER,
+        MULTIPLE_INTRO_OFFERS,
+        SELECTED,
+        UNSUPPORTED,
+    }
+
+    private object ConditionSerializer : KSerializer<Condition> {
```

Comment:
I was a bit confused with this one... The backend right now returns an object with a `type` property but in iOS, we deserialize that directly to the `Condition` enum so I followed the same pattern... I'm not sure if we add more properties to that object, if we would be able to expose it with this API though...